src/routeGeneration/templates/express.ts

```diff
@@ -76,23 +76,42 @@ export function RegisterRoutes(app: any) {
         return (request: any, response: any, next: any) => {
             let responded = 0;
             let success = false;
+
+            const succeed = function(user: any) {
+                if (!success) {
+                    success = true;
+                    responded++;
+                    request['user'] = user;
+                    next();
+                }
+            }
+
+            const fail = function(error: any) {
+                responded++;
+                if (responded == security.length && !success) {
+                    error.status = 401;
+                    next(error)
+                }
+            }
+
             for (const secMethod of security) {
-                expressAuthentication(request, secMethod.name, secMethod.scopes).then((user: any) => {
-                    // only need to respond once
-                    if (!success) {
-                        success = true;
-                        responded++;
-                        request['user'] = user;
-                        next();
+                if (Object.keys(secMethod).length > 1) {
+                    let promises: Promise<any>[] = [];
+
+                    for (const name in secMethod) {
+                        promises.push(expressAuthentication(request, name, secMethod[name]));
                     }
-                })
-                .catch((error: any) => {
-                    responded++;
-                    if (responded == security.length && !success) {
-                        response.status(401);
-                        next(error)
+
+                    Promise.all(promises)
+                        .then((values) => { succeed(values[0]); })
```

Comment:
Is it possible that a more explicit name than _values_ is used there ? Or does _expressAuthentication_ returns something not that much defined ?

Reply:
It is actually open what `expressAuthentication` returns. Could potentially also be a token object for instance. But in practice it's probably most often a user and since that's also used in `succeed`, I'll fix it here.